modules/data-streams/src/javaRestTest/java/org/elasticsearch/datastreams/LogsDataStreamRestIT.java

```diff
@@ -333,6 +333,23 @@ public void testLogsTimeSeriesIndexModeSwitch() throws IOException {
         );
         assertDataStreamBackingIndexMode("logsdb", 0, DATA_STREAM_NAME);
 
+        putTemplate(client, "custom-template", LOGS_STANDARD_INDEX_MODE);
```

Comment:
This change is needed because the upgrade path in `DataStream#unsafeRollover(...)` no longer allows going from tsdb directly to logsdb and visa versa.

Reply:
Additionally upgrading an existing logsdb data stream to tsdb (and visa versa) is a pretty extreme use case to support.